### DISCUSSION TITLE
Using dark/light theme with SSR?

### DISCUSSION THREAD

**Discussion by wmonecke** [tamagui/tamagui#1182] [General]
Is there a way we can set the theme with SSR? Currently, we see a flicker where it renders the light theme and then replaces it with the correct value returned by `useThemeName` // dark or light

**Comment by GigaBigBoss** [comment]
Also have seen this - Attempted multiple fixes and was unable to prevent this. Always seems to start Dark.

**Comment by natew** [comment]
I think the website shows how to do this. We use `next-theme` which injects some stuff into the head to make it switch before render.

What I do see is that disabling JS it will stay dark. I can think of a fix - need to have some way for our `prefers-` media queries to be stronger by default and then for js to make them less strong on hydration. But it's a bit lower priority since you can avoid flicker with something like next-theme.
